### PR TITLE
Increase max number of vector dims to 2048

### PR DESCRIPTION
**Pull request by mayya-sharipova**
Currently Lucene limits the max number of vector dimensions to 1024. 
This commit overrides KnnFloatVectorField and KnnByteVectorField 
classes to increase the limit to 2048 for indexed vectors in ES.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapperTests.java [hunk 1]
```diff
@@ -493,6 +492,67 @@ public void testDocumentsWithIncorrectDims() throws Exception {
         }
     }
 
+    /**
+     * Test that max dimensions limit for float dense_vector field
+     * is 2048 as defined by {@link XKnnFloatVectorField}
+     */
+    public void testMaxDimsFloatVector() throws IOException {
```
Comment (javanna) [inline server/src/test/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapperTests.java:499]: This test verifies that we support 2048, but not necessarily that it's the max number of dimensions? Do we already have a test that verifies that we can't go higher than that?

File: server/src/test/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldTypeTests.java [hunk 2]
```diff
@@ -186,5 +206,21 @@ public void testByteCreateKnnQuery() {
 
         e = expectThrows(IllegalArgumentException.class, () -> cosineField.createKnnQuery(new byte[] { 0, 0, 0 }, 10, null, null));
         assertThat(e.getMessage(), containsString("The [cosine] similarity does not support vectors with zero magnitude."));
+
+        DenseVectorFieldType fieldWith2048dims = new DenseVectorFieldType(
```
Comment (javanna) [inline server/src/test/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldTypeTests.java:210]: should we have a separate test method for this?

**Review by javanna — APPROVED** [review]
LGTM

File: server/src/test/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapperTests.java (hunk 1 above)
Comment (mayya-sharipova) [reply server/src/test/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapperTests.java:515]: Yes, we have a test called `testDims()`, but before we only checked the limit non-indexed vectors. I've added the test for indexed vectors.

File: server/src/test/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldTypeTests.java (hunk 2 above)
Comment (mayya-sharipova) [reply server/src/test/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldTypeTests.java]: Addressed in https://github.com/elastic/elasticsearch/pull/95257/commits/1d4b838ca1c8494673a833fdad9285e2dec4beb2

**Comment by mayya-sharipova** [comment]
@javanna Thanks for the feedback.

**Pull request merged**